pkg/hintrunner/zero/zerohint_ec.go

```diff
@@ -694,3 +695,34 @@ func createComputeSlopeV1Hinter(resolver hintReferenceResolver) (hinter.Hinter,
 
 	return newComputeSlopeV1Hint(point0, point1), nil
 }
+
+func newEcMulInnerHint(scalar hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "EcMulInner",
+		Op: func(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+			//> memory[ap] = (ids.scalar % PRIME) % 2
+
+			scalarFelt, err := hinter.ResolveAsFelt(vm, scalar)
+			if err != nil {
+				return err
+			}
+			scalarBytes := scalarFelt.Bytes()
+
+			resultUint256 := new(uint256.Int).SetBytes(scalarBytes[:])
```

Comment:
Look good, I am wondering if we should make some new util out of felt modulo.

Reply:
This is the only hint needing it. So probably ok to skip it for now 🤔